octoprint_mrbeam/analytics/analytics_handler.py

```diff
@@ -360,16 +362,18 @@ def _event_disk_space(self):
 		except:
 			self._logger.exception('Exception when saving info about the disk space')
 
-	def _event_laserhead_info(self):
+	def event_laserhead_info(self):
 		try:
+			lh = self._laserheadHandler.get_current_used_lh_data()
+			settings = self._laserheadHandler.get_correction_settings()
 			laserhead_info = {
-				ak.LASERHEAD_SERIAL: _mrbeam_plugin_implementation.lh['serial'],
-				ak.POWER_65: _mrbeam_plugin_implementation.lh['p_65'],
-				ak.POWER_75: _mrbeam_plugin_implementation.lh['p_75'],
-				ak.POWER_85: _mrbeam_plugin_implementation.lh['p_85'],
-				ak.CORRECTION_FACTOR: _mrbeam_plugin_implementation.lh['correction_factor'],
-				ak.CORRECTION_ENABLED: _mrbeam_plugin_implementation.lh['correction_enabled'],
-				ak.CORRECTION_OVERRIDE: _mrbeam_plugin_implementation.lh['correction_factor_override'],
+				ak.LASERHEAD_SERIAL: lh['serial'],
+				ak.POWER_65: lh['info']['p_65'],
```

Comment:
please prepare for the case that there is not data at all. currently you would get a key error

Reply:
done